ResourceStatusSystem/Utilities/Synchronizer.py

```diff
@@ -55,6 +56,36 @@ def sync( self, _eventName, _params ):
     :return: S_OK
     '''
 
+    sesHosts = CSHelpers.getStorageElementsHosts()
+    if not sesHosts[ 'OK' ]:
+      return sesHosts
+    sesHosts = sesHosts[ 'Value' ]
+
+    resources = sesHosts
+
+    ftsServer = getFTS3Servers()
+    if ftsServer[ 'OK' ]:
+      resources.extend( ftsServer[ 'Value' ] )
+
+    ce = CSHelpers.getComputingElements()
+    if ce[ 'OK' ]:
+      resources.extend( ce[ 'Value' ] )
+
+    downtimes = self.rManagement.selectDowntimeCache()
+
+    if not downtimes['OK']:
+      return downtimes
+
+    # Remove hosts that no longer exist in the CS
+    for host in downtimes['Value']:
+      gLogger.verbose('Checking if %s is still in the CS' % host[0] )
+      if host[0] not in resources:
+        gLogger.verbose( '%s is no longer in CS, removing entry...' % host[0] )
+        result = self.rManagement.deleteDowntimeCache( name = host[0] )
```

Comment:
Actually, the first time I tested it the table `DowntimeCache` in my local DB was empty. In this case nothing happens because the `downtimes['Value']` would be empty and the for loop will not be executed.